### PR TITLE
[SYCLomatic] Remove functions with the same function as isDefaultStream

### PR DESCRIPTION
**Pull request by wenhuiNi**
Signed-off-by: Ni, Wenhui <wenhui.ni@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/Utility.cpp [hunk 1]
```diff
@@ -3118,23 +3087,29 @@ bool isModifiedRef(const clang::DeclRefExpr *DRE) {
...
+  if (auto Paren = dyn_cast<ParenExpr>(Arg)) {
+    Arg = Paren->getSubExpr()->IgnoreImpCasts();
+  }
+  if (auto TypeCast = dyn_cast<ExplicitCastExpr>(Arg)) {
+    Arg = TypeCast->getSubExpr()->IgnoreImpCasts();
+  }
+  Expr::EvalResult Result;
+  if (!Arg->isValueDependent() &&
+      Arg->EvaluateAsInt(Result, dpct::DpctGlobalInfo::getContext())) {
+    auto Val = Result.Val.getInt().getZExtValue();
```
Comment (yihwang-nv) [inline clang/lib/DPCT/Utility.cpp]: You can directly use APSInt to do the comparison

File: clang/lib/DPCT/Utility.cpp [hunk 2]
```diff
@@ -3118,23 +3087,29 @@ bool isModifiedRef(const clang::DeclRefExpr *DRE) {
...
+    return Val < 3; // 0 or 1 (cudaStreamLegacy) or 2 (cudaStreamPerThread)
```
Comment (yihwang-nv) [inline clang/lib/DPCT/Utility.cpp]: Result.Val.getInt() < APSInt::get(3);

**Review by yihwang-nv — APPROVED** [review]
LGTM

File: clang/lib/DPCT/Utility.cpp [hunk 3]
```diff
@@ -3117,22 +3086,27 @@ bool isModifiedRef(const clang::DeclRefExpr *DRE) {
...
+  if (auto TypeCast = dyn_cast<ExplicitCastExpr>(Arg)) {
+    Arg = TypeCast->getSubExpr()->IgnoreImpCasts();
+  }
+  Expr::EvalResult Result;
+  if (!Arg->isValueDependent() &&
+      Arg->EvaluateAsInt(Result, dpct::DpctGlobalInfo::getContext())) {
+    return Result.Val.getInt() < APSInt::get(3); // 0 or 1 (cudaStreamLegacy) or 2 (cudaStreamPerThread)
+                    // all migrated to default queue;
+  }
+  if (dyn_cast<GNUNullExpr>(Arg->IgnoreImplicit()) ||
```
Comment (ziranzha) [inline clang/lib/DPCT/Utility.cpp]: Is NULL expr will be evaluated as 0 in above if stmt?

File: clang/lib/DPCT/Utility.cpp [hunk 4]
```diff
@@ -3117,22 +3086,27 @@ bool isModifiedRef(const clang::DeclRefExpr *DRE) {
...
-  } else if (Arg->getStmtClass() == Expr::CXXNullPtrLiteralExprClass) {
-    IsDefaultStream = true;
-  }
-  return IsDefaultStream;
+    return true;
+  }
+  if (auto Paren = dyn_cast<ParenExpr>(Arg)) {
+    Arg = Paren->getSubExpr()->IgnoreImpCasts();
+  }
+  if (auto TypeCast = dyn_cast<ExplicitCastExpr>(Arg)) {
```
Comment (ziranzha) [inline clang/lib/DPCT/Utility.cpp]: This may not need since it is ignored in line 3088

**Review by ziranzha — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**